cmdlaunch/cmdlaunch.py

```diff
@@ -46,16 +49,19 @@ def __init__(self, master):
         # pretty(self.icons)
         # Twice the loop to preserve image reference
         for i, icon in enumerate(self.icons):
-            self.x = Button(root, text=icon.info['name'] +'\n' + icon.info['version'], 
+            self.x = tk.Button(root, text=icon.info['name'] +'\n' + icon.info['version'], 
                         image=icon.photo,
-                        compound=LEFT,
-                        command=lambda icon=icon: self.button_exec(icon))
-            self.x.grid(row=0, column=i)
+                        compound=TOP,
+                        command=lambda icon=icon: self.button_exec(icon), width = 100,height =100)
+
+            self.x.grid(row=i // self.columns, column=i % self.columns)
 
     def button_exec(self, icon):
         os.chdir('programs/'+icon.program)
         for command in icon.info['commands']:
-            os.system(command)
+            command = 'gnome-terminal --command \"bash -c \\\"'+ command+'; exec bash\\\"\"'
```

Comment:
euhh what about windows?

Reply:
I am sorry, wasnot aware that tk is available on windows. Will do the needful. How do i cancel the PR ?

Reply:
No need just apply a cross-platform solution

Reply:
aslo i just merged a PR, you adjust

Reply:
Did not find any cross platform solution to achieve this. Any thoughts?

Reply:
I think the best you can do is put in different solutions for OS X, Windows and Linux (You can check the system with the sys module).
Found a related SO question which pretty much states that a general solution isn't possible:
https://stackoverflow.com/questions/19308415/execute-terminal-command-from-python-in-new-terminal-window

Reply:
@Zernerus ok

@Abhilash-Kar0  have an internal api like:

launch_window(program)

then we do the dirty logic inside that func

Reply:
Yeah @Zernerus saw that earlier.
@Abdur-rahmaanJ  I am trying that. Anyone can help me with mac ?

Reply:
@Abhilash-Kar0 oh just leave out mac for now!